src/include/migraphx/program.hpp

```diff
@@ -86,10 +86,12 @@ struct program
 
     instruction_ref add_outline(const shape& s);
 
-    instruction_ref add_parameter(std::string name, shape s);
+    instruction_ref add_parameter(std::string name, shape s, bool is_input = true);
```

Comment:
Can these changes be removed?

Reply:
We need this to indicate the scratch memory is not a input, so we can get correct input parameter names...

Reply:
We shouldn't rely on these parameter names as this is fragile. Can't we keep track of the parameter names before we compile?

Reply:
fixed